### PR TITLE
Use the hyphened --include-paths stancflag for 2.24+

### LINKED ISSUE #245: --include-paths warning in cmdstan 2.24
**Describe the bug**

Trying out cmdstan 2.24.0 rc1, I get the warning "--include_paths is deprecated. Please use --include-paths."

**To Reproduce**
First, install [cmdstan 2.24.0-rc1](https://github.com/stan-dev/cmdstan/releases/tag/v2.24.0-rc1)

Then:
```r
library(cmdstanr)
file <- file.path(cmdstan_path(), "examples", "bernoulli", "bernoulli.stan")
mod <- cmdstan_model(file, include_paths=".", force_recompile=TRUE)
#> Compiling Stan program...
#> --include_paths is deprecated. Please use --include-paths.
#> Then lots of compiler warnings, probably unrelated

```

**Expected behavior**
No `include_paths` warning.

**Operating system**
Ubuntu 20.04

**CmdStanR version number**
 0.0.0.9008 (e33e2b4)

**Additional context**
Minor request: it would be convenient if `include_paths` included the directory of the model file, as mentioned in https://github.com/stan-dev/cmdstanr/issues/59#issuecomment-549008393, but it's not a big deal.

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

Cmdstan 2.24 deprecated --include_paths in favor of --include-paths. So we should use that. Fixes #245 

The issue also has a request: 
"It would be convenient if include_paths included the directory of the model file"

This PR does not address that. I am not sure what would be best there. I guess there might be situation where users don't want "." in include paths.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar, Univ. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by jgabry** [comment]
> The issue also has a request:
> "It would be convenient if include_paths included the directory of the model file"
> 
> This PR does not address that. I am not sure what would be best there. I guess there might be situation where users don't want "." in include paths.

I think that most times they would want `"."` included but you're right that there could be cases where they don't. I'm not sure what to do about that either.

**Comment by jgabry** [comment]
Also, should we update the tests to use the release candidate instead of the latest release?

**Comment by rok-cesnovar** [comment]
> I'm not sure what to do about that either.

I think we could have the logic: "if no paths set use '.', else use whatever set". But that should be fixed at the stanc3 level.

> Also, should we update the tests to use the release candidate instead of the latest release?

Sure yeah. Will add that here.

**Comment by jgabry** [comment]
Actually maybe `cmdstan_default_path()` should be changed to not remove release candidates when setting the default path? 

The way it is currently 

https://github.com/stan-dev/cmdstanr/blob/e33e2b46828c3c72cd1325d9e27caa2828d6a27e/R/path.R#L114-L115

it will ignore the RC if an earlier release is present and, more importantly, if only the RC is installed the path won't get set automatically at all, and that could lead to errors.

**Comment by rok-cesnovar** [comment]
Yes, I agree. Just need to double check that if both X and X-rc are present it doesnt pickup the RC. Will fix tomorrow.

**Comment by jgabry** [comment]
Ok cool, thanks!

**Comment by jgabry** [comment]
@rok-cesnovar Do you know why the tests are failing or want help looking into it?

**Comment by rok-cesnovar** [comment]
I think I do know, just haven't had the time to look into this. Will do today. Cmdstan 2.24 no longer needs the precompiled main.o as it now takes care of it, the test is however written in such a way that it expects. Should be a simple fix.

**Comment by rok-cesnovar** [comment]
@jgabry fixed now. I left the tests for 2.23 in, though we could also delete them as I imagine we will always test with the latest cmdstan version, otherwise, it will become a maintenance burden.

File: R/model.R [hunk 1]
```diff
@@ -448,7 +454,9 @@ compile_method <- function(quiet = TRUE,
     }
   }
   stancflags_val <- paste0("STANCFLAGS += ", stancflags_val, paste0(stanc_built_options, collapse = " "))
-  prepare_precompiled(cpp_options, quiet)
+  if (cmdstan_version() < "2.24") {
+    prepare_precompiled(cpp_options, quiet)
```
Comment (rok-cesnovar) [inline R/model.R:458]: We can leave the `prepare_precompiled` function in for now, but I would maybe remove it after a release cycle, so when 2.25 is out? What do you think?

This function just makes versions 2.23 and earlier a bit more user friendly, but everything still works even without it, though a bit more care needs to be taken when using the thread,mpi and opencl flags with it.

So the function will become more or less dead code once users move to 2.24.

**Comment by rok-cesnovar** [comment]
Now using the freshly released 2.24 instead of the RC.

File: R/model.R (hunk 1 above)
Comment (jgabry) [reply R/model.R:458]: Sure, that makes. Let's remove it once 2.25 comes out. Should we open an issue to remind ourselves?

**Comment by jgabry** [comment]
Looks good!

> I think I do know, just haven't had the time to look into this.

Thanks, didn't mean to rush you! 

> I left the tests for 2.23 in, though we could also delete them as I imagine we will always test with the latest cmdstan version, otherwise, it will become a maintenance burden.

Yeah deleting them is fine. Do you want to do that before merging?

**Comment by rok-cesnovar** [comment]
> Do you want to do that before merging?

Done. I also bumped the version, 0.0.0.9010 is now recommended for use with > 2.23.

**Comment by jgabry** [comment]
Great, thanks! I assume the checks will pass but let's let them run and then we can merge.

**Comment by rok-cesnovar** [comment]
Yes, lets wait. Also added the link to the new install instructions, so we can close https://github.com/stan-dev/cmdstanr/issues/248

**Comment by jgabry** [comment]
Ok great thanks

**Comment by jgabry** [comment]
Do the new install instructions replace the blog post we've been linking to, or do we need both?

**Comment by rok-cesnovar** [comment]
The blog post probably no longer needs to be linked no.

**Pull request merged**